src/Spice86/Program.cs

```diff
@@ -41,56 +71,223 @@ public class Program {
+
+        BiosEquipmentDeterminationInt11Handler biosEquipmentDeterminationInt11Handler = new BiosEquipmentDeterminationInt11Handler(memory, cpu, loggerService);
+        SystemBiosInt12Handler systemBiosInt12Handler = new SystemBiosInt12Handler(memory, cpu, biosDataArea, loggerService);
+        SystemBiosInt15Handler systemBiosInt15Handler = new SystemBiosInt15Handler(memory, cpu, a20Gate, loggerService);
+        SystemClockInt1AHandler systemClockInt1AHandler = new SystemClockInt1AHandler(memory, cpu, loggerService, timerInt8Handler);
+        
+        EmulatorStateSerializer emulatorStateSerializer = new(configuration, memory, state, callbackHandler, executionFlowRecorder, functionHandler, loggerService);
+        
+        MainWindowViewModel? mainWindowViewModel = null;
+        ClassicDesktopStyleApplicationLifetime? desktop = null;
+        MainWindow? mainWindow = null;
+        ITextClipboard? textClipboard = null;
+        IHostStorageProvider? hostStorageProvider = null;
+        if (!configuration.HeadlessMode) {
+            desktop = CreateDesktopApp();
+            PerformanceViewModel performanceViewModel = new(state, pauseHandler);
+            mainWindow = new() {
+                PerformanceViewModel = performanceViewModel
+            };
+            textClipboard = new TextClipboard(mainWindow.Clipboard);
+            hostStorageProvider = new HostStorageProvider(mainWindow.StorageProvider, configuration, emulatorStateSerializer);
+            mainWindowViewModel = new MainWindowViewModel(
+                timer, new UIDispatcher(Dispatcher.UIThread), hostStorageProvider, textClipboard, configuration, loggerService, pauseHandler);
         }
-    }
+        
+        using (mainWindowViewModel) {
+            VgaCard vgaCard = new(mainWindowViewModel, vgaRenderer, loggerService);
+            Keyboard keyboard = new Keyboard(state, ioPortDispatcher, a20Gate, dualPic, loggerService, mainWindowViewModel, configuration.FailOnUnhandledPort);
+            BiosKeyboardInt9Handler biosKeyboardInt9Handler = new BiosKeyboardInt9Handler(memory, cpu, dualPic, keyboard, biosDataArea, loggerService);
+            Mouse mouse = new Mouse(state, dualPic, mainWindowViewModel, configuration.Mouse, loggerService, configuration.FailOnUnhandledPort);
+            
+            MouseDriver mouseDriver = new MouseDriver(cpu, memory, mouse, mainWindowViewModel, vgaFunctionality, loggerService);
+
+            KeyboardInt16Handler keyboardInt16Handler = new KeyboardInt16Handler(memory, cpu, loggerService, biosKeyboardInt9Handler.BiosKeyboardBuffer);
+            Dos dos = new Dos(memory, cpu, keyboardInt16Handler, vgaFunctionality, configuration.CDrive,
+                configuration.Exe, configuration.Ems,
+                new Dictionary<string, string>() { { "BLASTER", soundBlaster.BlasterString } },
+                loggerService);
+            
+            if (configuration.InitializeDOS is not false) {
```

Comment:
Do we still need/want this config?

Cause with this MR a couple of things are initialized that were not initialized before when this was false.
And also, aren't  a couple of the interrupts that get installed here not even part of DOS?

Reply:
This should be équivalent to master as far as the usage of initializeDos flag is concerned.

The setting itself is misnamed. It's used to prevent the initialization of many things of we boot up a PC booter or bios. It goes beyond DOS.

Reply:
It's mainly the Default filehandles and the env vars.

Reply:
Good catch. I fixed it in the DOS class.